lib/gen_stage.ex

```diff
@@ -882,11 +882,17 @@ defmodule GenStage do
 
   @callback init(args :: term) ::
               {:producer, state}
+              | {:producer, state, :hibernate | {:continue, term}}
```

Comment:
The current PR doesn't handle hiberrnate, does it? In any case, there are other discussions we need to have regarding hibernate, so probably better to focus on continue for now. :)

Reply:
No, but the existing documentation for the init function mentions hibernate, which confused me. I'll remove the hibernate reference from the docs (around lines 833-835) and from this callback.

Reply:
>  I'll remove the hibernate reference from the docs (around lines 833-835) and from this callback.

Fantastic, thank you!

Reply:
Hmm, actually, the `handle_info` callback (and the others) also contains `:hibernate`. I guess that was a copy paste from the GenServer docs?

Reply:
Good catch! Actually we do support hibernate there but not on `init`. So I guess we crossed this line anyway. So I would go ahead and support it on `init` too!